### PR TITLE
move addNewDataField/removeExistingDataField to tool and make it handle fieldcollections/objectbricks and classes

### PR DESCRIPTION
**Pull request by AlternateIf**
Tested/used the addNewDataField/removeExistingDataField function in Classdefinition to implement a tool that handles insertion (back/front), appending/prepending, replacing and removing in a tool

This tool works for fieldcollection, class, and objectbrick definitions

functionality detail:

appendFields: adds an element/elements after a given element by name
prependFields: adds an element/elements before a given element by name
insertFieldsFront: inserts an element/elements into a panel (front)
insertFieldsBack: inserts an element/elements into a panel (back)
replaceField: replaces a give element with an element/elements
removeField: removes a given element

Note that save has to be called outside of the tool (same behaviour as in the former functions in ClassDefinition)

### CODE REVIEW

**Comment by AlternateIf** [comment]
@fashxp

**Comment by dpfaffenbauer** [comment]
Thats a BC.

**Comment by AlternateIf** [comment]
true thanks :) @dpfaffenbauer

File: pimcore/models/DataObject/ClassDefinition.php [hunk 1]
```diff
@@ -1135,10 +1135,11 @@ public function getLinkGenerator()
     }
 
     /**
+     * @deprecated Just a BC compatibility method
```
Comment (fashxp) [inline pimcore/models/DataObject/ClassDefinition.php]: can't that be delegated to the new helper class?

**Comment by fashxp** [comment]
Still Work in Progress - please do not merge yet.

**Comment by fashxp** [comment]
thx very much

**Pull request merged**